src/components/Dialog/stories/Modal.stories.tsx

```diff
@@ -33,3 +49,61 @@ Default.args = {
     onOpening: fn(),
     onClosing: fn(),
 };
+
+/** Nested modals with modal context for tracking open/close states. */
+const InnerModal = () => {
+    const [isOpen, setIsOpen] = React.useState(true);
+
+    return (
+        <Modal isOpen={isOpen} usePortal={true} size={"tiny"} modalId={"innerModal"}>
+            <TrackingContent
+                more={
+                    <>
+                        <Spacing key={"spacing"} />
+                        <Button key={"close"} onClick={() => setIsOpen(false)} size={"md"}>
+                            Close
+                        </Button>
+                    </>
+                }
+            />
+        </Modal>
+    );
+};
+
+const MiddleModal = () => {
+    return (
+        <Modal isOpen={true} modalId={"middleModal"} size={"small"} usePortal={true}>
+            <TrackingContent more={<InnerModal />} />
+        </Modal>
+    );
+};
+
+/** Shows the current stack of open modals. */
+const TrackingContent = ({ more }: { more: React.JSX.Element }) => {
+    const modalContext = React.useContext(ModalContext);
+
+    return (
+        <Card>
+            <CardHeader key="1">
+                <CardTitle>Tracking content</CardTitle>
+            </CardHeader>
+            <CardContent>
+                <ul>
+                    {(modalContext.openModalStack ?? []).map((modalId, idx) => (
+                        <li key={modalId}>
+                            {idx + 1}. {modalId}
+                        </li>
+                    ))}
+                </ul>
+                {more}
+            </CardContent>
+        </Card>
+    );
+};
+
+export const NestedModalWithContext = ContextTemplate.bind({});
```

Comment:
This story made the docs page inaccessible, see https://62150dbccb2d77003a2a5093-zmchvutxgz.chromatic.com/?path=/docs/components-dialog-modal--docs